In [223]:
from bs4 import BeautifulSoup
import requests
import re

In [225]:
url='https://www.linkedin.com/feed/update/urn:li:activity:7234945468089171968/'

page=requests.get(url)

soup = BeautifulSoup(page.content, 'html.parser')

In [230]:
print(soup)

<!DOCTYPE html>

<html lang="en">
<head>
<meta content="d_public_post" name="pageKey"/>
<!-- --><!-- --> <meta content="en_US" name="locale"/>
<meta data-app-version="1.2.640" data-browser-id="ad669349-8dbc-403e-8138-4ea95cae2c24" data-call-tree-id="AAYhHwWwRkc5rRFeigw4Tg==" data-dfp-member-lix-treatment="control" data-disable-jsbeacon-pagekey-suffix="false" data-dna-member-lix-treatment="control" data-enable-page-view-heartbeat-tracking="" data-human-member-lix-treatment="control" data-member-id="0" data-multiproduct-name="feedcontent-guest-frontend" data-page-instance="urn:li:page:d_public_post;CtbQASNMRTir9s47Iegk9g==" data-service-name="feedcontent-guest-frontend" id="config"/>
<link href="https://www.linkedin.com/posts/ozibook_internship-hiringinterns-dataanalystinterns-activity-7234945468089171968-gJH4" rel="canonical"/>
<!-- --><!-- -->
<!-- -->
<!-- -->
<meta content="https://www.linkedin.com/posts/ozibook_internship-hiringinterns-dataanalystinterns-activity-7234945468089171968

In [231]:
result = soup.find('div', class_='details mx-details-container-padding')

In [232]:
result

<div class="details mx-details-container-padding">
<section class="mb-3">
<h1 class="section-title mb-2">
                Ozibook’s Post
              </h1>
<article class="relative pt-1.5 px-2 pb-0 bg-color-background-container container-lined main-feed-activity-card main-feed-activity-card-with-comments" data-activity-urn="urn:li:activity:7234945468089171968" data-attributed-urn="urn:li:share:7234592466501308417" data-featured-activity-urn="urn:li:activity:7234945468089171968">
<!-- -->
<div class="flex items-center font-sans mb-1" data-test-id="main-feed-activity-card__entity-lockup">
<a class="relative" data-tracking-control-name="public_post_feed-actor-image" data-tracking-will-navigate="" href="https://in.linkedin.com/company/ozibook?trk=public_post_feed-actor-image">
<img alt="View organization page for Ozibook, graphic" class="inline-block relative w-6 h-6" data-delayed-url="https://media.licdn.com/dms/image/v2/C4D0BAQFTRBnbsN0yCA/company-logo_100_100/company-logo_100_100/0/165

In [229]:
span_data = result.find_all('span')

In [233]:
print(span_data)

[<span class="!text-xs text-color-text-low-emphasis leading-[1.33333] flex">
<time class="flex-none">
                  

    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    

    
    
    
    
    
    
    
    
    
    
    
    
    
    

      3d
  
<!-- --> </time>
<!-- --> </span>, <span aria-hidden="true" class="font-normal ml-0.5" data-test-id="social-actions__reaction-count">
                    15
              </span>, <span class="social-action-bar__button-text">
      Like
    </span>, <span class="social-action-bar__button-text">
      Comment
    </span>, <span class="social-share__button-text-square social-action-bar__button-text">
              Share
            </span>, <span class="social-share__item-text pl-1">Copy</span>, <span class="social-share__item-text pl-1">LinkedIn</span>, <span class="social-share__item-text pl-1">Facebook</span>, <span class="social-share__item-text pl-1">Twitter</span>, <span class="text-xs text-color-text

In [235]:
word_span_data = [span.text.strip() for span in span_data]

In [217]:
n = word_span_data[0]

In [236]:
print("Post uploaded "+n+" before")

Post uploaded 3d before


## The post as posted at 3 days before today 2 sept 2024 then the date of post uploded on 30 Aug 2024

In [237]:
post_data = result.find_all('a')

In [222]:
post_data

[<a class="relative" data-tracking-control-name="public_post_feed-actor-image" data-tracking-will-navigate="" href="https://in.linkedin.com/company/ozibook?trk=public_post_feed-actor-image">
 <img alt="View organization page for Ozibook, graphic" class="inline-block relative w-6 h-6" data-delayed-url="https://media.licdn.com/dms/image/v2/C4D0BAQFTRBnbsN0yCA/company-logo_100_100/company-logo_100_100/0/1653577608837/ozibook_logo?e=2147483647&amp;v=beta&amp;t=XGo-EXlRYYuOoNfuhyW4KSX-624Pso_WP0dUvIhcWAs" data-ghost-classes="bg-color-entity-ghost-background" data-ghost-url="https://static.licdn.com/aero-v1/sc/h/cs8pjfgyw96g44ln9r7tct85f"/>
 <!-- --> </a>,
 <a aria-label="View organization page for Ozibook" class="text-sm link-styled no-underline leading-open" data-tracking-control-name="public_post_feed-actor-name" data-tracking-will-navigate="" href="https://in.linkedin.com/company/ozibook?trk=public_post_feed-actor-name">
               Ozibook
             </a>,
 <a class="semaphore__tog

In [195]:
post_data[17]

<a aria-label="15 Reactions" class="flex items-center font-normal text-color-text-low-emphasis no-underline visited:text-color-text-low-emphasis my-1" data-id="social-actions__reactions" data-num-reactions="15" data-plural="%numReactions%" data-separate-ctas="false" data-singular="%numReactions%" data-test-id="social-actions__reactions" data-tracking-control-name="public_post_social-actions-reactions" data-tracking-will-navigate="" href="https://www.linkedin.com/signup/cold-join?session_redirect=https%3A%2F%2Fwww%2Elinkedin%2Ecom%2Fposts%2Fozibook_internship-hiringinterns-dataanalystinterns-activity-7234945468089171968-gJH4&amp;trk=public_post_social-actions-reactions" tabindex="0" target="_self">
<img alt="" data-delayed-url="https://static.licdn.com/aero-v1/sc/h/9sun4diznbhgyelaeryxy75ur" data-reaction-type="LIKE" height="16px" width="16px"/>
<span aria-hidden="true" class="font-normal ml-0.5" data-test-id="social-actions__reaction-count">
                    15
              </span>

In [201]:
Likes = [span.text.strip() for span in post_data[17] if span.text.strip()]

In [206]:
print("Likes =",Likes)

Likes = ['15']


## The post has 15 likes/Reactions as of September 2, 2024, 12:30 PM.

In [207]:
post_data[18]

<a class="flex items-center font-normal text-color-text-low-emphasis no-underline visited:text-color-text-low-emphasis before:middot my-1" data-id="social-actions__comments" data-num-comments="106" data-plural="%numComments% Comments" data-separate-ctas="false" data-singular="%numComments% Comment" data-test-id="social-actions__comments" data-tracking-control-name="public_post_social-actions-comments" data-tracking-will-navigate="" href="https://www.linkedin.com/signup/cold-join?session_redirect=https%3A%2F%2Fwww%2Elinkedin%2Ecom%2Fposts%2Fozibook_internship-hiringinterns-dataanalystinterns-activity-7234945468089171968-gJH4&amp;trk=public_post_social-actions-comments" target="_self">
        
                106 Comments
            
      </a>

In [209]:
column_span_data = [a.text.strip() for a in post_data[18]]

In [212]:
print("Comments =",column_span_data)

Comments = ['106 Comments']


## The post has 106 comments as of September 2, 2024, 12:30 PM.

In [238]:
print("Post uploaded "+n+" before")
print("Likes =",Likes)
print("Comments =",column_span_data)

Post uploaded 3d before
Likes = ['15']
Comments = ['106 Comments']
